### PR TITLE
add import service

### PR DESCRIPTION
**Pull request by alexeyMohnatkin**
1 - File serverless.yml contains configuration for importProductsFile function **DONE**
3 - The importProductsFile lambda function returns a correct response which can be used to upload a file into the S3 bucket **DONE**
[https://03gweslkx1.execute-api.eu-west-1.amazonaws.com/dev/import](https://03gweslkx1.execute-api.eu-west-1.amazonaws.com/dev/import)
4 - Frontend application is integrated with importProductsFile lambda **DONE**
[https://d21tkn9gbb6vp3.cloudfront.net/admin/products](https://d21tkn9gbb6vp3.cloudfront.net/admin/products)
Pull request: [https://github.com/alexeyMohnatkin/nodejs-aws-fe/pull/5](https://github.com/alexeyMohnatkin/nodejs-aws-fe/pull/5)
5 - The importFileParser lambda function is implemented and serverless.yml contains configuration for the lambda **DONE**

Additional (optional) tasks
+1 - async/await is used in lambda functions **DONE**
+1 - importProductsFile lambda is covered by unit tests (aws-sdk-mock can be used to mock S3 methods - https://www.npmjs.com/package/aws-sdk-mock) **NOT_DONE**
+1 - At the end of the stream the lambda function should move the file from the uploaded folder into the parsed folder (move the file means that file should be copied into parsed folder, and then deleted from uploaded folder) **DONE**

### CODE REVIEW

File: import-service/serverless.ts [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+    'serverless-offline',
+  ],
+  provider: {
+    name: 'aws',
+    runtime: 'nodejs12.x',
+    apiGateway: {
+      minimumCompressionSize: 1024,
+    },
+    environment: {
+      AWS_NODEJS_CONNECTION_REUSE_ENABLED: '1',
```
Comment (helly-fox) [inline import-service/serverless.ts:30]: You can also move bucket name here and use it as process.env

**Review by helly-fox — APPROVED** [review]

File: import-service/handlers/importFileParser.ts [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+        console.log(data);
+
+        results.push(data)
+      })
+      .on('end', () => {
+        console.log(results);
+        console.log('end');
+      });
+
+    await s3.copyObject({
```
Comment (SkorykV) [inline import-service/handlers/importFileParser.ts]: tell me, if I'm wrong, but this code will not wait untill file is parsed.

Comment (sleepwalky) [reply import-service/handlers/importFileParser.ts]: indeed, copying and deleting should be invoked in .on('end', () => {}) callback

File: import-service/serverless.ts [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+    importProductsFile: {
+      handler: 'handler.importProductsFile',
+      events: [
+        {
+          http: {
+            method: 'get',
+            path: 'import',
+            cors: {
+              origins: '*',
+            },
```
Comment (EugeneCh) [inline import-service/serverless.ts:53]: no query parameter described here

Comment (alexeyMohnatkin) [reply import-service/serverless.ts:53]: sorry, for what purpose?

File: import-service/handlers/importFileParser.ts (hunk 2 above)
Comment (alexeyMohnatkin) [reply import-service/handlers/importFileParser.ts]: thanks for noticing, fixed

File: import-service/serverless.ts (hunk 3 above)
Comment (EugeneCh) [reply import-service/serverless.ts:53]: Well, as said in our task "The name of CSV file with products will be passed in a query string as a name parameter and should be described in the serverless.yml file as a request parameter."

**Pull request merged**